src/prpy/planning/ompl.py

```diff
@@ -245,3 +246,100 @@ def ShortcutPath(self, robot, path, timeout=1., **kwargs):
+        @param planner_range the max stepsize for the planner before a projection
+         is applied
+        @param ompl_args ompl RRTConnect specific parameters
+        @return traj
+        """
+        if ompl_args is None:
+            ompl_args = {}
+        ompl_args['range'] = planner_range
+        return self._TSRPlan(robot, tsrchains, ompl_args=ompl_args, **kw_args)
+
+    @ClonedPlanningMethod
+    def PlanToEndEffectorOffset(self, robot, direction, distance, planner_range=0.1,
+                                smoothingitrs=100, **kw_args):
+        """
+        Plan to a desired end-effector offset.
+        @param robot
+        @param direction unit vector in the direction of motion
+        @param distance minimum distance in meters
+        @param planner_range the max stepsize for the planner before a projection
+         is applied
+        @param smoothingitrs number of smoothing iterations to run
+        @return traj output path
+        """
+        direction = numpy.array(direction, dtype=float)
+
+        if direction.shape != (3,):
+            raise ValueError('Direction must be a three-dimensional vector.')
+        if not (distance >= 0):
+            raise ValueError('Distance must be non-negative; got {:f}.'.format(
+                             distance))
+
+        with robot:
+            manip = robot.GetActiveManipulator()
+            H_world_ee = manip.GetEndEffectorTransform()
+
+            # 'object frame w' is at ee, z pointed along direction to move
+            import prpy
+            H_world_w = prpy.kin.H_from_op_diff(H_world_ee[0:3,3], direction)
+            H_w_ee = numpy.dot(prpy.kin.invert_H(H_world_w), H_world_ee)
+
+            # Serialize TSR string (whole-trajectory constraint)
+            Bw = numpy.zeros((6,2))
+            epsilon = 0.00
+            Bw = numpy.array([[-epsilon,            epsilon],
+                              [-epsilon,            epsilon],
+                              [min(-epsilon, distance-epsilon),  
+                                max(epsilon, distance+epsilon)],
+                              [-epsilon,            epsilon],
+                              [-epsilon,            epsilon],
+                              [-epsilon,            epsilon]])
+
+            from prpy.tsr.tsr import TSR, TSRChain
+            trajtsr = TSR(T0_w = H_world_w, 
+                          Tw_e = H_w_ee, 
+                          Bw = Bw, 
+                          manip = robot.GetActiveManipulatorIndex())
+
+            traj_tsr_chain = TSRChain(constrain=True, TSRs=[trajtsr])
+
+            # Compute a goal pose
+            goal_pose = manip.GetEndEffectorTransform()
+            goal_pose[:3,3] += direction*distance/numpy.linalg.norm(direction)
+
+            # Get all IKs for the goal pose and send them all to the planner
+            goal_configs = manip.FindIKSolutions(goal_pose, 0)
```

Comment:
I thought `or_ompl` supported goal sets. Why do we manually find IK solutions here?